### ISSUE TITLE
[Test-Proxy] Add more support for Uri sanitization

### ISSUE THREAD

**Issue by JoshLove-msft** [Azure/azure-sdk-tools#2368]
Right now you can use the UriRegexSanitizer, but ideally there would be a simple sanitizer that accepts the query param keys to sanitize. This could be used for both the request URI as well as request body for content-type of application/x-www-form-urlencoded.

There should also be a simple way to sanitize the other parts of the URI
- host
- path
- query params

**Comment by JoshLove-msft** [comment]
To be clear there is a viable work around for this right now.

**Comment by JoshLove-msft** [comment]
As a concrete example, here is what we would have to do right now to sanitize a host name:

```c#
UriRegexSanitizers.Add(
    new UriRegexSanitizer(@"^((http[s]?|ftp):\/)?\/?(?<host>[^:\/\s]+)((\/\w+)*\/)([\w\-\.]+[^#?\s]+)(.*)?(#[\w\-]+)?$", FAKE_HOST)
    {
        GroupForReplace = "host"
    });
```

**Comment by pakrym** [comment]
Urls are not secrets we need to stop doing this.

**Comment by kurtzeborn** [comment]
I agree with Pavel here.  Closing.

CC: @scbedd

**Comment by weshaggard** [comment]
@pakrym is there no case where secrets might show up in the uri? I'm thinking cases like a SAS token as part of the uri?

**Comment by pakrym** [comment]
Yes, they can be in the query (sanitizing which we already support) but never in the path.

**Comment by pakrym** [comment]
Actually, I might be wrong. @JoshLove-msft  @scbedd do we support query parameter sanitization?

**Comment by JoshLove-msft** [comment]
> Actually, I might be wrong. @JoshLove-msft @scbedd do we support query parameter sanitization?

There is a UriRegexSanitizer which lets you apply a regex to the entire URI. The point of this issue was to ask for an easier way to handle sanitizing the various parts of the URI. E.g. for query params, a key could be specified.

**Comment by kurtzeborn** [comment]
Assigning to @scbedd for consideration...
